packages/gatsby-plugin-google-analytics/src/gatsby-node.js

```diff
@@ -1,3 +1,50 @@
+// Copied from https://gist.github.com/faisalman/924970#file-analytics-regexp-js
+// NOTE: This is just an example, in many cases we'd want to async check some API
+async function isTrackingId(str) {
+  return /^ua-\d{4,9}-\d{1,4}$/i.test(str)
+}
+
+exports.pluginOptionsSchema = ({ Joi }) =>
```

Comment:
Probably not worth it, but wondering if a less tool oriented alias might help us. "Validator" or something, so that we could decouple things if needed.

Reply:
I second this. If for whatever reason we would have to drop `Joi` in future, that name would be somewhat confusing if it's no longer actually `Joi`.

The separate problem with above is that if we drop `Joi` we would still be tied to `Joi` API and whatever replacement would be, it would need to match the API

Reply:
I think if we use Joi we should call it Joi so folks who have questions can find the plethora of answers that already exist for the library. Calling it anything else would be more confusing than anything.

We _could_ also try to come up with our own library/API instead of Joi, but that seems like a lot of bikeshedding for... no real benefit?

Reply:
The other option is not to provide anything other than hook to run whatever validation plugin would use. We can still document it with `joi` package, but plugins would need to bring their own. This minimizes API surface for us significantly?

`hapi` recently just did that - they no longer expose `joi` in public API and instead users have to import `joi` themselves (and put it in dependencies).

Reply:
We need to know the plugin schema to allow us to build interactive forms for configuring plugins, both in the CLI and Admin. Sorry for not specifying that in the FAQ, great point @pieh!

Reply:
I have to agree with Max here, the reason we are opinionated is that we can do other optimizations under the hood. We've been using joi for a very long time and it's used very broadly. I also think making it explicit is great.